AutoPartsShop.API/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,140 @@
+﻿using AutoPartsShop.API.Extensions;
+using AutoPartsShop.API.Repositories.Interfaces;
+using AutoPartsShop.Models.DTOs;
+using Microsoft.AspNetCore.Mvc;
+
+namespace AutoPartsShop.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ProductsController : ControllerBase
+    {
+        private readonly IProductRepository productRepository;
+
+
```

Comment:
Usually you need only one new line between methods, variables declaration and other code parts. Please double check all the files and make sure you don't have unnecessary new lines like this one.